cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java

```diff
@@ -262,21 +262,15 @@ public UUID getUserTaskId(HttpServletRequest httpServletRequest) {
     return userTaskId;
   }
 
-  private synchronized boolean isActiveUserTasksDone(UUID userTaskId) {
-    UserTaskInfo userTaskInfo = _allUserTaskIdToFutureMap.get(TaskState.ACTIVE).get(userTaskId);
-    if (userTaskInfo == null || userTaskInfo.futures().isEmpty()) {
-      return true;
-    }
-
-    List<OperationFuture> futures = userTaskInfo.futures();
-    return futures.get(futures.size() - 1).isDone();
-  }
-
    synchronized void checkActiveUserTasks() {
     Iterator<Map.Entry<UUID, UserTaskInfo>> iter = _allUserTaskIdToFutureMap.get(TaskState.ACTIVE).entrySet().iterator();
     while (iter.hasNext()) {
       Map.Entry<UUID, UserTaskInfo> entry = iter.next();
-      if (isActiveUserTasksDone(entry.getKey())) {
+      if (entry.getValue().isUserTaskDoneExceptionally()) {
+        LOG.warn("UserTask {} is complete with Exception and removed from active tasks list", entry.getKey());
```

Comment:
Nit: `complete with Exception ` -> `completed with exception ` (let's fix the `complete` -> `completed` at `else if...` part)